cartoframes/data/registry/query_dataset.py

```diff
@@ -24,12 +26,29 @@ def query(self):
 
     @property
     def dataset_info(self):
-        raise CartoException('We can not extract Dataset info from a QueryDataset. Use a TableDataset '
-                             '`Dataset(table_name)` to get or modify the info from a CARTO table.')
-
-    def update_dataset_info(self, privacy=None, name=None):
-        raise CartoException('We can not extract Dataset info from a QueryDataset. Use a TableDataset '
-                             '`Dataset(table_name)` to get or modify the info from a CARTO table.')
+        if self._dataset_info is None:
+            tables = self._get_tables_used_by_query()
+            if not tables or len(tables) == 0:
+                raise CartoException('We can not extract the table/s used by the QueryDataset, so we can not get the '
+                                     'Dataset info. Try using `Dataset(table_name)` instead if possible.')
+            elif len(tables) == 1:
+                self._dataset_info = self._get_dataset_info(tables[0])
+            else:
+                self._dataset_info = {}
```

Comment:
We should be consistent with the return type.

I prefer returning always a dictionary even when there's just one result, than having to figure out which is the response type anytime there's a call to this method.

Reply:
:+1: